src/sentry/static/sentry/app/components/resultGrid.tsx

```diff
@@ -129,103 +142,113 @@ class SortBy extends React.Component {
-      endpoint: '',
-      method: 'GET',
-      columns: [],
-      sortOptions: [],
-      filters: {},
-      defaultSort: '',
-      keyForRow: function (row) {
-        return row.id;
-      },
-      columnsForRow: function () {
-        return [];
-      },
-      defaultParams: {
-        per_page: 50,
-      },
-      hasPagination: true,
-      hasSearch: false,
-    };
-  },
-
-  getInitialState() {
-    const queryParams = (this.props.location || {}).query || {};
+type FilterConfig = {
+  name: string;
+  options: Option[];
+};
+
+type Props = {
+  api: Client;
+  location: Location;
+} & DefaultProps;
+
+type DefaultProps = {
+  columns: React.ReactNode[];
+  columnsForRow: (row: any) => React.ReactNode[];
+  defaultSort: string;
+  defaultParams: {[k: string]: any};
+  filters: {[k: string]: FilterConfig};
+  endpoint: string;
+  hasPagination: boolean;
+  hasSearch: boolean;
+  keyForRow: (row: any) => string;
+  method: RequestOptions['method'];
+  path: string;
+  sortOptions: Option[];
+};
+
+type State = {
+  rows: any[];
+  loading: boolean;
+  error: string | boolean;
+  pageLinks: null | string;
+  query: string;
+  sortBy: string;
+  filters: {[x: string]: string};
+};
+
+class ResultGrid extends React.Component<Props, State> {
+  static defaultProps: DefaultProps = {
+    path: '',
+    endpoint: '',
+    method: 'GET',
+    columns: [],
+    sortOptions: [],
+    filters: {},
+    defaultSort: '',
+    keyForRow: row => row.id,
+    columnsForRow: () => [],
+    defaultParams: {
+      per_page: 50,
+    },
+    hasPagination: true,
+    hasSearch: false,
+  };
 
-    return {
-      rows: [],
-      loading: true,
-      error: false,
-      pageLinks: null,
-      query: queryParams.query || '',
-      sortBy: queryParams.sortBy || this.props.defaultSort,
-      filters: Object.assign({}, queryParams),
-    };
-  },
+  state: State = this.defaultState;
 
   componentWillMount() {
     this.fetchData();
-  },
+  }
 
-  componentWillReceiveProps(nextProps) {
-    const queryParams = (nextProps.location || {}).query || {};
+  componentWillReceiveProps() {
+    const queryParams = this.query;
     this.setState(
       {
-        query: queryParams.query || '',
-        sortBy: queryParams.sortBy || this.props.defaultSort,
+        query: queryParams.query ?? '',
+        sortBy: queryParams.sortBy ?? this.props.defaultSort,
         filters: Object.assign({}, queryParams),
         pageLinks: null,
         loading: true,
         error: false,
       },
       this.fetchData
     );
-  },
+  }
+
+  get defaultState() {
+    const queryParams = this.query;
+
+    return {
+      rows: [],
+      loading: true,
+      error: false,
+      pageLinks: null,
+      query: queryParams.query ?? '',
+      sortBy: queryParams.sortBy ?? this.props.defaultSort,
+      filters: Object.assign({}, queryParams),
+    } as State;
+  }
+
+  get query() {
+    return ((this.props.location ?? {}).query ?? {}) as {[k: string]: string};
```

Comment:
```suggestion
    return ((this.props.location ?? {}).query ?? {}) as Record<string, string>;
```

Reply:
Any reason to not use the `Query` type from the location package?